File: app/src/main/java/net/itsjustsomedude/tokens/db/AppDatabase.java
package net.itsjustsomedude.tokens.db;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;

import android.content.Context;

@Database(entities = {Coop.class, Event.class}, version = 1)
@TypeConverters(Converters.class)
public abstract class AppDatabase extends RoomDatabase {
	private static AppDatabase instance;

	public abstract CoopDao coopDao();

	public abstract EventDao eventDao();

	public static synchronized AppDatabase getInstance(Context context) {
		if (instance == null) {
			instance = Room.databaseBuilder(context.getApplicationContext(),
							AppDatabase.class, "database")
					.fallbackToDestructiveMigration()
					.build();
		}
		return instance;
	}
}

File: app/src/main/java/net/itsjustsomedude/tokens/db/AppDatabase.kt
package net.itsjustsomedude.tokens.db

import android.content.Context
import androidx.room.Database
import androidx.room.Room.databaseBuilder
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(
	entities = [Coop::class, Event::class],
	version = 2,
	exportSchema = true
)
@TypeConverters(Converters::class)
abstract class AppDatabase : RoomDatabase() {
	abstract fun coopDao(): CoopDao

	abstract fun eventDao(): EventDao

	companion object {
		fun createInstance(context: Context): AppDatabase = databaseBuilder(
			context.applicationContext,
			AppDatabase::class.java,
			"Coops.db"
		)
			// Migrates from the legacy Tokifiction to the Room version.
			.addMigrations(MIGRATION_10_1)
			// Adds boost order to Coop, and Receiver to Events.
			.addMigrations(MIGRATION_1_2)
//            .fallbackToDestructiveMigration()
			.build()
	}
}